geode-core/src/integrationTest/java/org/apache/geode/internal/cache/DiskRegionCompactorClearOplogAfterRecoveryJUnitTest.java

```diff
@@ -0,0 +1,190 @@
+    await().untilAsserted(() -> assertThat(getCurrentNumberOfOplogs(diskStore)).isEqualTo(5));
+
+    Set<Long> oplogIds = getAllOplogIds(diskStore);
+
+    region.close();
+    region = createRegion();
+
+    // Execute destroy operations for all entries created in previous step. All oplogs created
+    // in previous step will not contain live entries, so they must be compacted.
+    executeDestroyOperations(region);
+
+    await().untilAsserted(
+        () -> assertThat(areOplogsCompacted(oplogIds, diskStore)).isTrue());
+  }
+
+  boolean areOplogsCompacted(Set<Long> oplogIds, DiskStoreImpl diskStore) {
+    Set<Long> currentOplogId = getAllOplogIds(diskStore);
+    return currentOplogId.stream().noneMatch(oplogIds::contains);
+  }
+
+  Set<Long> getAllOplogIds(DiskStoreImpl diskStore) {
+    Set<Long> oplogIds = new HashSet<>();
+    for (Oplog oplog : diskStore.getAllOplogsForBackup()) {
+      oplogIds.add(oplog.getOplogId());
+    }
+    return oplogIds;
+  }
+
+
+  void executePutOperations(Region<Object, Object> region) {
+    for (int i = 0; i < ENTRY_RANGE; i++) {
+      region.put(i, new byte[100]);
+    }
+  }
+
+  void executeDestroyOperations(Region<Object, Object> region) throws InterruptedException {
+    TombstoneService tombstoneService = ((InternalCache) cache).getTombstoneService();
+    for (int i = 0; i < ENTRY_RANGE; i++) {
+      region.destroy(i);
+      assertThat(tombstoneService.forceBatchExpirationForTests(1)).isTrue();
+    }
+  }
+
+  void createDiskStore(int compactionThreshold, int maxOplogSizeInBytes) {
+    DiskStoreFactory diskStoreFactory = cache.createDiskStoreFactory();
+    diskStoreFactory.setAutoCompact(true);
+    diskStoreFactory.setCompactionThreshold(compactionThreshold);
+    diskStoreFactory.setDiskDirsAndSizes(diskDirs, diskDirSizes);
+
+    createDiskStoreWithSizeInBytes(diskStoreName, diskStoreFactory, maxOplogSizeInBytes);
+  }
+
+  Region<Object, Object> createRegion() {
+    RegionFactory<Object, Object> regionFactory =
+        cache.createRegionFactory(RegionShortcut.PARTITION_PERSISTENT);
+    regionFactory.setDataPolicy(DataPolicy.PERSISTENT_PARTITION);
+    regionFactory.setDiskStoreName(diskStoreName);
+    regionFactory.setDiskSynchronous(true);
+    return regionFactory.create(regionName);
+  }
+
+  int getCurrentNumberOfOplogs(DiskStoreImpl ds) {
+    return ds.getAllOplogsForBackup().length;
+  }
+
+  private File createDirectory(File parentDirectory, String name) {
+    File file = new File(parentDirectory, name);
+    assertThat(file.mkdir()).isTrue();
+    return file;
+  }
+
+  private void createDiskStoreWithSizeInBytes(String diskStoreName,
+      DiskStoreFactory diskStoreFactory,
+      long maxOplogSizeInBytes) {
+    ((DiskStoreFactoryImpl) diskStoreFactory).setMaxOplogSizeInBytes(maxOplogSizeInBytes);
+    ((DiskStoreFactoryImpl) diskStoreFactory).setDiskDirSizesUnit(DiskDirSizesUnit.BYTES);
+    diskStoreFactory.create(diskStoreName);
+  }
```

Comment:
I would change this method to require `DiskStoreFactoryImpl diskStoreFactory` as a parameter just to be a bit cleaner.